akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala

```diff
@@ -785,6 +795,13 @@ abstract class PersistentActorSpec(config: Config) extends PersistenceSpec(confi
       expectMsg("x-b-2")
       expectNoMsg(100.millis)
     }
+    "support calling persistAll with Nil" in {
+      val persistentActor = namedPersistentActor[PersistAllNilPersistentActor]
+      persistentActor ! Cmd("x")
+      expectMsg("before-nil")
+      expectMsg("after-nil")
+      expectMsg("x")
```

Comment:
Ah, I wasn’t consciously aware of the optimization within `deferAsync` to immediately execute when possible—dare I say that this might be confusing?